VUModManagerRegistry/Services/S3/S3ModStorage.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Threading.Tasks;
+using Amazon.S3;
+using Amazon.S3.Model;
+using Microsoft.Extensions.Options;
+using VUModManagerRegistry.Common.Interfaces;
+
+namespace VUModManagerRegistry.Services.S3
+{
+    [ExcludeFromCodeCoverage]
+    public sealed class S3ModStorageOptions
+    {
+        public string ServiceURL { get; set; }
+        
+        [NotNull]
+        public string BucketName { get; set; }
+    }
+    
+    [ExcludeFromCodeCoverage]
+    public class S3ModStorage : IModStorage
+    {
+        private readonly string _bucketName;
+        private readonly AmazonS3Client _s3Client;
+
+        public S3ModStorage(IOptions<S3ModStorageOptions> options)
+        {
+            _bucketName = options.Value.BucketName;
```

Comment:
whats this IOptions thingy and why do you need that?

Reply:
It's something from ASP.net to inject options from your settings file